### PR TITLE
Mc 1500 fix engagement count

### PR DESCRIPTION
**Pull request by chelseyklein**
## Goal
What changed? What is the business/product goal?

- We are seeing double engagement counts due to events being calculated twice. These changes filter out records where `scheduled_corpus_item_id` is null to filter out merino data in legacy queries

## References

JIRA ticket:
[MC-1500](https://mozilla-hub.atlassian.net/browse/MC-1500)

Screenshot 2024-09-12 at 1 28 17 PM
Screenshot 2024-09-12 at 1 27 45 PM

[MC-1500]: https://mozilla-hub.atlassian.net/browse/MC-1500?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: data-products/src/firefox_newtab/sql/glean_firefox_new_tab_impressions_hourly/glean_firefox_new_tab_daily_engagement_by_tile_id_position/data.sql [hunk 1]
```diff
@@ -48,6 +48,9 @@ WITH
         'recommendation_id') IS NOT NULL
       OR mozfun.map.get_key(e.extra,
         'tile_id') IS NOT NULL)
+    --filter out records where scheduled corpus item ID is populated to prevent double count
+    AND mozfun.map.get_key(e.extra,
+        'scheduled_corpus_item_id') IS NULL
```
Comment (chidamnat) [inline data-products/src/firefox_newtab/sql/glean_firefox_new_tab_impressions_hourly/glean_firefox_new_tab_daily_engagement_by_tile_id_position/data.sql:54]: what about not null scenario. where is that counted ?

Comment (chelseyklein) [reply data-products/src/firefox_newtab/sql/glean_firefox_new_tab_impressions_hourly/glean_firefox_new_tab_daily_engagement_by_tile_id_position/data.sql:54]: That's happening [here](https://github.com/Pocket/data-flows/blob/c88aac8df8b8fe1bda351048671a723d0760efce/data-products/src/firefox_newtab/sql/glean_firefox_new_tab_impressions_hourly/glean_firefox_new_tab_daily_engagement_by_scheduled_corpus_item_id_position_country_locale/data.sql#L47) - I believe the three of these are joined (or unioned) in the downstream dbt models

File: data-products/src/firefox_newtab/sql/glean_firefox_new_tab_impressions_hourly/glean_firefox_new_tab_daily_engagement_by_tile_id_position/data.sql [hunk 2]
```diff
@@ -48,6 +48,9 @@ WITH
         'recommendation_id') IS NOT NULL
       OR mozfun.map.get_key(e.extra,
         'tile_id') IS NOT NULL)
+    --filter out records where scheduled corpus item ID is populated to prevent double count
```
Comment (gkatre) [inline data-products/src/firefox_newtab/sql/glean_firefox_new_tab_impressions_hourly/glean_firefox_new_tab_daily_engagement_by_tile_id_position/data.sql:51]: Perhaps a mention that this filter: includes data prior to Firefox 129

**Review by gkatre — APPROVED** [review]

**Review by chidamnat — APPROVED** [review]

File: data-products/src/firefox_newtab/sql/glean_firefox_new_tab_impressions_hourly/glean_firefox_new_tab_daily_engagement_by_tile_id_position/data.sql (hunk 1 above)
Comment (gkatre) [reply data-products/src/firefox_newtab/sql/glean_firefox_new_tab_impressions_hourly/glean_firefox_new_tab_daily_engagement_by_tile_id_position/data.sql:54]: @chidamnat in addition to what Chelsey mentioned, this change relates to only pre-merino engagement data.

For Firefox 129 and later (Merino based recommendations), the data extracts are happening in another Prefect flow (as Chelsey mentioned above).

Finally in the data models we are merging the engagement data from the new Merino and prior data in the data model `firefox_new_tab_daily_engagement_by_scheduled_corpus_item_id_position_source_country_locale_x`, the one that you have used recently to update a metaflow job

**Pull request merged**